.github/workflows/release.yaml

```diff
@@ -31,3 +43,20 @@ jobs:
             ./bin/release/storageos_linux_amd64.sha256
             ./bin/release/storageos_windows_amd64
             ./bin/release/storageos_windows_amd64.sha256
+
+  dockerhub-release:
+    runs-on: ubuntu-18.04
+    name: Publish container to Dockerhub
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Login to container registry
+        uses: docker/login-action@v1
+        with:
+          registry: docker.io
+          username: ${{ secrets.DOCKERHUB_CLI_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_CLI_PASSWORD }}
+      - name: Build image
+        run: SEMANTIC_VERSION=${{ env.SEMANTIC_VERSION }} make build-cli-image
+      - name: Push image
+        run: SEMANTIC_VERSION=${{ env.SEMANTIC_VERSION }} make push-cli-image
```

Comment:
Should we just run this as an extra step to the `Publish release binaries`. As otherwise the tests may fail, yet we still publish an image to DH.

It also stops us building the image twice. Not sure what's idiomatic for Github actions

Reply:
No, good point, i've added `needs` statements to make the prerequisites explicit 

I'm not sure about building the image twice though, as far as I can see the publish-binaries step doesn't build the CLI image, just the binaries (albeit doing it within a container)

Reply:
> I'm not sure about building the image twice though, as far as I can see the publish-binaries step doesn't build the CLI image, just the binaries (albeit doing it within a container)

Yep, you're right. I misread it. 

Looks good, thanks!